### PR TITLE
Update to base 4.16.1.0

### PR DESCRIPTION
**Pull request by mxxun**
(And also through all intermediate bases)

This hopefully gets right the list of modules in base-4.16; it's surprisingly nonobvious _which_ base.cabal must contain the list as of 4.16.

### CODE REVIEW

**Comment by pasqu4le** [comment]
So. for context, this was opened for [morley/#784](https://gitlab.com/morley-framework/morley/-/issues/784).

There we use `stack` and I see that currently the `base` version `4.15.1.0` is the newest one used by a [stackage LTS](https://www.stackage.org/lts-19.6/package/base-4.15.1.0).

I suppose that we want to use that version now and eventually `4.16.1.0` after a stackage LTS adopts it. 

I think we can split this PR in two commits where the first one uses `4.15.1.0`, so that we can pull either of these two from `master` once this is merged.

**Comment by mxxun** [comment]
> I think we can split this PR in two commits where the first one uses `4.15.1.0`, so that we can pull either of these two from `master` once this is merged.

As it is now, there are four commits; for `4.15.0.0`, `4.15.1.0`, `4.16.0.0`, and `4.16.1.0`, just in case. Do you mean to squash two intermediate commits, or to split this PR in two, or..?

**Comment by pasqu4le** [comment]
Oh, right :facepalm: 

Either I was very distracted or GitHub played some trick on me, because I totally didn't notice the 4 commits.

> Do you mean to squash two intermediate commits

No, what you did is perfectly sane, I just didn't see it for some reason.
Sorry about that, please disregard my previous comment.

File: base-noprelude.cabal
Comment (pasqu4le) [inline base-noprelude.cabal]: it seems that in some of these commits there are some `exposed-modules` from `base` that are missing here
e.g. in a1606be67e7ea64ccd41e978dfde54d9c5c7c8e6 I don't see:
- `GHC.Event.TimeOut`
- `GHC.IO.StdHandles`
- `GHC.IO.SubSystem`
- `GHC.Integer`
- `GHC.Integer.Logarithms`
- `GHC.IOPort`

nor some OS-specific ones (see at the bottom).

I'm mentioning this here because of this comment as well, because I think that it's easier to grab the list of modules from `hackage` itself, by visiting the:
> Package description (as included in the package)

link, aka:
```
https://hackage.haskell.org/package/base-<version>/base.cabal
```
for each `<version>` and looking for the `exposed-modules` there.

File: base-noprelude.cabal
Comment (pasqu4le) [inline base-noprelude.cabal]: ah, I see that this comes from the `cabal` file for `base`, but I think that we can remove this comment

File: base-noprelude.cabal
Comment (mxxun) [reply base-noprelude.cabal]: Indeed! Fixed now, I think.

File: base-noprelude.cabal
Comment (mxxun) [reply base-noprelude.cabal]: (Done)

File: base-noprelude.cabal
Comment (pasqu4le) [reply base-noprelude.cabal]: just checked, AFAICT everything appears to be present.

I see you have also included the `reexported-modules` from `base`, good call :ok_hand:

File: base-noprelude.cabal
Comment (pasqu4le) [reply base-noprelude.cabal]: Nitpick: I'm un-resolving this simply because I see that there's now an empty line (with spaces).

I'd prefer that to be no line at all, as it seems to be the case for previous commits :slightly_smiling_face:

**Review by pasqu4le — APPROVED** [review]
I only (re-)left one nitpick above, but aside from that all LGTM.

I'm approving already, feel free to merge this once that change is made :+1:

File: base-noprelude.cabal
Comment (mxxun) [reply base-noprelude.cabal]: Ooook; dropped.

**Pull request merged**
